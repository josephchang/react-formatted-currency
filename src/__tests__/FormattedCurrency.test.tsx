import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CurrencyParts } from 'format-currency-to-parts';
import FormattedCurrency, { IFormattedCurrencyProps, RenderArgs } from '../';

describe('<FormattedCurrency />', () => {
  let props: IFormattedCurrencyProps;
  let wrapper: ShallowWrapper<IFormattedCurrencyProps>;

  describe('GIVEN GBP 43210.96', () => {
    beforeEach(() => {
      props = {
        currency: 'GBP',
        amount: 43210.96,
        locale: 'en-gb',
        parts: [CurrencyParts.INTEGER, CurrencyParts.DECIMALS],
        render: (parts: RenderArgs) => (
          <div>
            <span>{parts.integer}</span>
            <sup>{parts.decimals}</sup>
          </div>
        ),
      };

      wrapper = shallow(<FormattedCurrency {...props} />);
    });

    it('should render the formatted currency', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
