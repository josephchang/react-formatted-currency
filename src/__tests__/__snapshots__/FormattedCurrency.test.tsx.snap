// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormattedCurrency /> GIVEN GBP 43210.96 should render the formatted currency 1`] = `
<div>
  <span>
    43,210
  </span>
  <sup>
    .96
  </sup>
</div>
`;
